### ISSUE TITLE
training model failed when setting poolSize and filtSize to non-default values

### ISSUE THREAD

**Issue by EMIAOZANG** [EMIAOZANG/deeplearning#3]
I was trying different poolSize and filtSize values, somehow I got this error when poolSize and filtSize not equal to default values:
# e.g. when filtSize=5, poolSize=10, I got the following error message:

filtSize=5
poolSize=10
==> construct model
==> here is the model:
nn.Sequential {
  [input -> (1) -> (2) -> (3) -> (4) -> (5) -> (6) -> (7) -> (8) -> (9) -> (10) -> (11) -> (12) -> output](1): nn.SpatialConvolutionMM(1 -> 64, 5x5)
  (2): nn.Tanh
  (3): nn.Sequential {
    [input -> (1) -> (2) -> (3) -> (4) -> output](1): nn.Square
    (2): nn.SpatialAveragePooling(10,10,10,10)
    (3): nn.MulConstant
    (4): nn.Sqrt
  }
  (4): nn.SpatialSubtractiveNormalization
  (5): nn.SpatialConvolutionMM(64 -> 64, 5x5)
  (6): nn.Tanh
  (7): nn.Sequential {
    [input -> (1) -> (2) -> (3) -> (4) -> output](1): nn.Square
    (2): nn.SpatialAveragePooling(10,10,10,10)
    (3): nn.MulConstant
    (4): nn.Sqrt
  }
  (8): nn.SpatialSubtractiveNormalization
  (9): nn.Reshape(1600)
  (10): nn.Linear(1600 -> 128)
  (11): nn.Tanh
  (12): nn.Linear(128 -> 10)
}
==> define loss
==> here is the loss function:
nn.ClassNLLCriterion
==> defining some tools
==> configuring optimizer
==> defining training procedure
==> defining test procedure
==> training!
==> doing epoch on training data:
==> online epoch # 1 [batchSize = 1]
/Users/luchristopher/torch/install/bin/luajit: ...hristopher/torch/install/share/lua/5.1/nn/Sequential.lua:44: Given input size: (64x2x2). Calculated output size: (64x-2x-2). Output size is too small at /Users/luchristopher/torch/extra/nn/generic/SpatialConvolutionMM.c:78
stack traceback:
    [C]: in function 'updateOutput'
    ...hristopher/torch/install/share/lua/5.1/nn/Sequential.lua:44: in function 'forward'
    4_train.lua:160: in function 'opfunc'
    .../luchristopher/torch/install/share/lua/5.1/optim/sgd.lua:44: in function 'optimMethod'
    4_train.lua:184: in function 'train'
    doall.lua:80: in main chunk
    [C]: in function 'dofile'
    ...pher/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
#     [C]: at 0x0108fe7bb0

It seems that the training process only works when poolSize and filtSize are set to default values, we might need to debug this part tomorrow.

**Comment by charlesdguthrie** [comment]
Ok I'm looking into this now
